pkg/surveyutils/jsonschema.go

```diff
@@ -771,12 +769,27 @@ func (o *JSONSchemaOptions) handleBasicProperty(name string, prefixes []string,
 		}
 		if o.VaultClient != nil {
 			dereferencedFormat := util.DereferenceString(t.Format)
-			path := strings.Join([]string{o.VaultBasePath, strings.Join(prefixes, "-")}, "/")
+			// lets ignore the last prefix if its the same as the key
+			// to avoid redundant path
+			pathPrefixes := prefixes
+			lastIdx := len(prefixes) - 1
+			if prefixes[lastIdx] == dereferencedFormat {
+				pathPrefixes = prefixes[0:lastIdx]
+			}
+			path := strings.Join([]string{o.VaultBasePath, strings.Join(pathPrefixes, "-")}, "/")
 			secretReference := secreturl.ToURI(path, dereferencedFormat, o.VaultScheme)
```

Comment:
I think it's better to just use `prefixes[lastIdx]` as the key, rather than doing this dance of seeing if it's the same... Just so that we don't have to guess at what's happening with the paths

Reply:
good call! another commit removes format as the key and uses the last path...